src/ClientData/include/ClientData/ScopeId.h

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_CLIENT_DATA_API_SCOPE_ID_H_
+#define ORBIT_CLIENT_DATA_API_SCOPE_ID_H_
+
+#include "OrbitBase/Typedef.h"
+
+namespace orbit_client_data {
+
+struct ScopeIdTag {};
+
+using ScopeId = orbit_base::Typedef<ScopeIdTag, const uint64_t>;
+
+static_assert(sizeof(ScopeId) == sizeof(uint64_t), "orbit_base::Typedef is not zero-cost");
```

Comment:
Checking size is only half of it. You have to check alignment as well to make this useful.